### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by mit-mit**
Add details for how to use the image

Moved out from now abandoned https://github.com/dart-lang/dart-docker/pull/8

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,6 +9,52 @@ images as well as for information regarding contributing and issues.
...
+The following `Dockerfile` performs two steps:
+
+1. Using the Dart SDK in the `dart:stable` image, compiles your server
+(`bin/server.dart`) to an executable (`server`).
+
+1. Assembles the runtime image by combining the compiled server with the Dart VM
+runtime and it's needed dependencies located in `/runtime/`.
+
+```
+# Use latest stable channel SDK.
```
Comment (subfuzion) [inline README.md:27]: See the wording I used [here](https://github.com/dart-lang/dart-docker/blame/794ca592448d784ef40ad773f3f2b6217720a2b7/README.md#L12) for this. I also should have made it clearer that for production we recommend explicitly specifying the desired version. We should make it clear here or in the bullet above that we support two channels (`dart:<version>` as `dart:stable` or `dart:beta` will pull the latest in that channel). For production, we recommend the `stable` channel and using explicit version tags (generally prefer the most explicit version) to ensure deterministic behavior when rebuilding an image. Ex: `dart:2.12.4`. (Might be too much detail here to note that users should also consider pinning to specific digests in production. Details [here](https://docs.docker.com/engine/reference/commandline/pull/#pull-an-image-by-digest-immutable-identifier)).

File: README.md [hunk 2]
```diff
@@ -9,6 +9,52 @@ images as well as for information regarding contributing and issues.
...
+FROM scratch
+COPY --from=build /runtime/ /
+COPY --from=build /server /bin/
+
+# Start server.
+EXPOSE 8080
+CMD ["/bin/server"]
+```
+
+If you have [Docker Desktop][] installed, you can build and run with the
```
Comment (subfuzion) [inline README.md:50]: Same nit about  empty reference `[]`

File: README.md [hunk 3]
```diff
@@ -9,6 +9,52 @@ images as well as for information regarding contributing and issues.
...
+1. Assembles the runtime image by combining the compiled server with the Dart VM
+runtime and it's needed dependencies located in `/runtime/`.
+
+```
+# Use latest stable channel SDK.
+FROM dart:stable AS build
+
+# Resolve app dependencies.
+WORKDIR /app
+COPY pubspec.yaml .
```
Comment (subfuzion) [inline README.md:32]: Should be `COPY pubspec.* .` for apps (shown [here](https://github.com/dart-lang/dart-docker/blame/794ca592448d784ef40ad773f3f2b6217720a2b7/README.md#L17)).

File: README.md [hunk 4]
```diff
@@ -9,6 +9,52 @@ images as well as for information regarding contributing and issues.
 The [Docker Hub page] is generated/maintained over in the
 [docker-library/docs repository], specifically in [the `dart` directory].
 
+## Using this image
+
+We recommend creating small runtime images by leveraging Dart's support for
+ahead-of-time (AOT) [compilation to executables][]. This enables creating small
```
[the remark this answers is not in the document]
Comment (mit-mit) [reply README.md:15]: Done

File: README.md (hunk 2 above)
Comment (mit-mit) [reply README.md:50]: Done

File: README.md (hunk 3 above)
Comment (mit-mit) [reply README.md:32]: Done

File: README.md (hunk 1 above)
Comment (mit-mit) [reply README.md:27]: I see, so there essentially isn't a lock file (similar to `pubspec.lock`) for the image version, so if you build the image again, you might get a new version?

Comment (subfuzion) [reply README.md:27]: Yes. you can republish to the same tag multiple times. In practice, that normally won't happen for explicit tags, but publishers can and do republish to less specific tags all the time to ensure they're the latest (like to `stable` in our case, as well as to `latest`, etc.).

Comment (mit-mit) [reply README.md]: Got it, updated

**Comment by mit-mit** [comment]
@subfuzion I'm merging this, as I fixed your last comment. If we need adjustments, we can do another PR.

**Pull request merged**